sqlite3/src/lib.rs

```diff
@@ -918,3 +923,36 @@ fn sqlite3_errstr_impl(rc: i32) -> *const std::ffi::c_char {
         }
     }
 }
+
+#[no_mangle]
+pub unsafe extern "C" fn sqlite3_wal_checkpoint(
+    _db: *mut sqlite3,
+    _db_name: *const std::ffi::c_char,
+) -> ffi::c_int {
+    sqlite3_wal_checkpoint_v2(
+        _db,
+        _db_name,
+        SQLITE_CHECKPOINT_PASSIVE,
+        std::ptr::null_mut(),
+        std::ptr::null_mut(),
+    )
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn sqlite3_wal_checkpoint_v2(
+    db: *mut sqlite3,
+    _db_name: *const std::ffi::c_char,
+    _mode: ffi::c_int,
+    _log_size: *mut ffi::c_int,
+    _checkpoint_count: *mut ffi::c_int,
+) -> ffi::c_int {
+    if db.is_null() {
+        return SQLITE_MISUSE;
+    }
+    let db: &mut sqlite3 = &mut *db;
+    // TODO: Checkpointing modes and reporting back log size and checkpoint count to caller.
+    if let Err(e) = db.conn.checkpoint() {
+        return SQLITE_ERROR;
+    }
+    SQLITE_OK
+}
```

Comment:
should we log what modes are not supported for now? I think currently the only used one is a bad version of truncate